src/transformers/quantizers/quantizer_quanto.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2024 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import TYPE_CHECKING, Optional
+
+from .base import HfQuantizer
+
+
+if TYPE_CHECKING:
+    from ..modeling_utils import PreTrainedModel
+
+from ..integrations import replace_with_quanto_linear
+from ..utils import is_quanto_available, is_torch_available, logging
+from ..utils.quantization_config import QuantoConfig
+
+
+if is_torch_available():
+    import torch
+
+logger = logging.get_logger(__name__)
+
+
+class QuantoHfQuantizer(HfQuantizer):
+    """
+    Quantizer for the quanto library
+    """
+
+    required_packages = ["quanto", "accelerate"]
+
+    def __init__(self, quantization_config: QuantoConfig, **kwargs):
+        self.requires_calibration = quantization_config.activations is not None
```

Comment:
For the transformers integration, we will only support quantization for weight only quantization since : 
- it does not require calibration data.
- Don't need to retrain most of the time. 
- We can perform the weight only quantization on the spot (just like bnb by quantizing each layer as we load them) -> memory efficient. 

However, for the activation, we need the whole model to be loaded and pass the calibration data. Hence, there is no real benefit of supporting the quantization of the activation in transformers. I think that it will be easier for the user to directly use quanto + perform QAT if needed. LMK if it makes sense ! If the user requests it, we can still add it. 
cc @younesbelkada

Reply:
yes, I think we can discuss about it but currently I think in terms of UX we should probably not silently calibrate the model by default, I agree we should do it only when users expliclty asks for it, what do you think @dacorvo ?
I think the main caveat of not calibrating the model under the hood is the amount of time it might take for large models to just load the model